askama_shared/src/generator.rs

```diff
@@ -803,24 +803,31 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         self.handle_ws(ws);
         self.write_buf_writable(buf)?;
         buf.write("let ");
+        self.visit_target(buf, var);
+        buf.writeln(";")
+    }
+
+    fn write_let_needs_let(&self, var: &'a Target) -> bool {
         match *var {
             Target::Name(name) => {
                 let name = normalize_identifier(name);
-                self.locals.insert_with_default(name);
-                buf.write(name);
-            }
-            Target::Tuple(ref targets) => {
-                buf.write("(");
-                for name in targets {
-                    let name = normalize_identifier(name);
-                    self.locals.insert_with_default(name);
-                    buf.write(name);
-                    buf.write(",");
+                match self.locals.get(&name) {
+                    None => {
+                        // declares a new variable
+                        true
+                    }
+                    Some(meta) if meta.initialized => {
+                        // an initialized variable gets shadowed
+                        true
+                    }
+                    _ => {
+                        // initializes a variable that was introduced in a LetDecl before
+                        false
+                    }
```

Comment:
Maybe this could/should be a method on `MapChain`, to be able to say `.is_*()`?

Reply:
Only if we can use it in multiple places. I would suggest moving the comments out of the match arms, which should make this a lot more compact.

Reply:
Fair, I guess it's a matter of preference. But yes, I agree with the moving the comments :)

Updated suggestion:

```suggestion
                match self.locals.get(&name) {
                    // declares a new variable
                    None => true,
                    // an initialized variable gets shadowed
                    Some(meta) if meta.initialized => true,
                    // initializes a variable that was introduced in a LetDecl before
                    _ => false,
```